pkg/queue/cluster_queue.go

```diff
@@ -171,6 +183,14 @@ func (c *ClusterQueue) PushOrUpdate(wInfo *workload.Info) {
 	c.heap.PushOrUpdate(wInfo)
 }
 
+func (c *ClusterQueue) Heapify(lqName string) {
+	for _, wl := range c.heap.List() {
```

Comment:
Hm, this should probably take a lock on the CQ, otherwise elements may be added / returned while iterating over the heap, no?

Reply:
Added a lock, however this function already locks the mutex on Manager before it's called. I agree with your suggestion, however let's consider in the future if we aren't a bit too conservative about locks given we have them both in Manager and CQ objects. We can maybe decrease the number of locks in the whole file in a follow up

Reply:
> Added a lock, however this function already locks the mutex on Manager before it's called. 

The question is if every access to the CQ goes via the manager, I remember analyzing the code in the past, and there were accesses bypassing the manager, but I'm not fully sure at the moment.

> We can maybe decrease the number of locks in the whole file in a follow up

maybe there is room for improvement, I cannot recall exactly the reason we introduced 2 levels of locking, most likely there were (maybe still are) some functions bypassing the manager.

Reply:
I agree, it requires some little investigation. I believe it's out of scope for this PR